### DISCUSSION TITLE
Relative Checkstyle

### DISCUSSION THREAD

**Discussion by tomerghelber-tm** [eslint/eslint#16793] [Q&A]
Hello,

We had an old problem with `checkstyle` formatter and jenkins, and we needed the result path to be relative for the plugin in Jenkins to show the currect code.

In eslint v8 the formatter became hidden and we cannot fix the issue anymore.

This is the fix:
```js
const checkstyle = require('eslint/lib/cli-engine/formatters/checkstyle')

function fixPath({ filePath, ...rest }) {
    return {
        filePath: filePath.replace(process.env.INIT_CWD, "").slice(1),
        ...rest
    }
}

module.exports = function(results, data) {
    const fixedPaths = results.map(fixPath)
    return checkstyle(fixedPaths)
}
```

Is there a replacement in eslint 8? Or maybe another fix?

I couldn't find anyone asks about it, sorry if this is a duplication.

Thanks.

**Comment by mdjermanovic** [comment]
Built-in formatters modules are not intended to be public. You can simply copy & paste [the code](https://github.com/eslint/eslint/blob/main/lib/cli-engine/formatters/checkstyle.js) into your custom formatter.
